File: src/python/roman-to-integer.py
# https://leetcode.com/problems/roman-to-integer/

class Solution:
    def romanToInt(self, s: str) -> int:
        roman_dic = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
        ans = 0

        for i in range(len(s)):

            if i != 0 and roman_dic[s[i - 1]] < roman_dic[s[i]]:
                ans = ans + roman_dic[s[i]] - roman_dic[s[i - 1]] * 2
            else:
                ans = ans + roman_dic[s[i]]

        return ans

File: src/cpp/roman-to-integer.cpp
class Solution {
int getTranslateNum(char s) {
    switch(s) {
        case 'I': return 1;
        case 'V': return 5;
        case 'X': return 10;
        case 'L': return 50;
        case 'C': return 100;
        case 'D': return 500;
        case 'M': return 1000;
        default: return 0;
    }
    return 0;
}
public:
    int romanToInt(string s) {
        if(s.empty()) return 0;
        int returnValue = 0;
        for(unsigned int i=0; i<s.size()-1; i++) {
            if(getTranslateNum(s[i]) < getTranslateNum(s[i+1])) {
                returnValue -= getTranslateNum(s[i]);
            } else {
                returnValue += getTranslateNum(s[i]);
            }
        }
        returnValue += getTranslateNum(s[s.size()-1]);
        return returnValue;
    }
};

// https://leetcode.com/submissions/detail/814767425/
